src/pages/gen2/start/quickstart/nextjs-app-router-server-components/index.mdx

```diff
@@ -0,0 +1,274 @@
+
+<Fragments fragments={{ javascript: buildABackend, nextjs: buildABackend }} />
+
+
+## Build UI
+
+Let's add UI that connects to the backend data and auth resources.
+
+
+### Configure Amplify Client Side
+
+First, install the Amplify UI component library:
+
+```bash
+npm install @aws-amplify/ui-react
+```
+
+Next, create a `components` folder in the root of your project and the contents below to a file called `ConfigureAmplify.tsx`.
+
+```ts title="components/ConfigureAmplify.tsx"
+// components/ConfigureAmplify.tsx
+"use client";
+
+import config from "@/amplifyconfiguration.json";
+import { Amplify } from "aws-amplify";
+
+Amplify.configure(config, { ssr: true });
+
+export default function ConfigureAmplifyClientSide() {
+  return null;
+}
+```
+
+Update `app/layout.tsx` to import and render `<ConfigureAmplifyClientSide />`.  This client component will configure Amplify for client pages in our application.
+
+```ts title="app/layout.tsx"
+// app/layout.tsx
+import ConfigureAmplifyClientSide from "@/components/ConfigureAmplify";
+import "@aws-amplify/ui-react/styles.css";
+import type { Metadata } from "next";
+import { Inter } from "next/font/google";
+import "./globals.css";
+
+const inter = Inter({ subsets: ["latin"] });
+
+export const metadata: Metadata = {
+  title: "Create Next App",
+  description: "Generated by create next app",
+};
+
+export default function RootLayout({
+  children,
+}: {
+  children: React.ReactNode;
+}) {
+  return (
+    <html lang="en">
+      <body className={inter.className}>
+        <ConfigureAmplifyClientSide />
+        {children}
+      </body>
+    </html>
+  );
+}
+```
+
+### Add a login page
+
+Create a new route under `app/login/page.tsx` to render the [Amplify Authenticator component](https://ui.docs.amplify.aws/react/connected-components/authenticator) when a user is not logged in or redirect to the root route.
+
+```ts title="app/login/page.tsx"
+// app/login/page.tsx
+
+"use client";
+
+import { withAuthenticator } from "@aws-amplify/ui-react";
+import { redirect } from "next/navigation";
+import { useEffect } from "react";
+
+function Login({ user }) {
+  useEffect(() => {
+    if (user) {
+      redirect("/");
+    }
+  }, [user]);
+  return null;
+}
+
+export default withAuthenticator(Login);
+```
+
+### Configure Amplify Server Side
+
+Next, create a `utils/amplify-utils.ts` file from the root of the project and paste the code below.  `runWithAmplifyServerContext` and `cookiesClient` are declared here and will be used to gain access to Amplify assets from the server.
+
+
+```ts title="utils/amplify-utils.ts"
+// utils/amplify-utils.ts
+import { type Schema } from "@/amplify/data/resource";
+import config from "@/amplifyconfiguration.json";
+import { createServerRunner } from "@aws-amplify/adapter-nextjs";
+import { generateServerClientUsingCookies } from "@aws-amplify/adapter-nextjs/api";
+import { cookies } from "next/headers";
```

Comment:
Can we restructure these imports so the local imports are underneath the lib imports, leading with the next provided imports, to create a mental model separating Amplify (lib and local) and NextJS specific imports:
```suggestion
import { cookies } from "next/headers";

import { createServerRunner } from "@aws-amplify/adapter-nextjs";
import { generateServerClientUsingCookies } from "@aws-amplify/adapter-nextjs/api";

import { type Schema } from "@/amplify/data/resource";
import config from "@/amplifyconfiguration.json";
```

Same question applies to all other import examples